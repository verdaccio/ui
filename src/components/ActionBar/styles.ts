import styled from 'react-emotion';
import { default as MuiFab } from '@material-ui/core/Fab';

import colors from '../../utils/styles/colors';
import ListItem from '../../muiComponents/ListItem';

export const ActionListItem = styled(ListItem)({
  paddingTop: 0,
  paddingLeft: 0,
  paddingRight: 0,
});

export const Fab = styled(MuiFab)({
  '&&': {
    backgroundColor: colors.primary,
    color: colors.white,
    marginRight: '10px',
  },
});
