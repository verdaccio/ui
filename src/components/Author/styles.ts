import styled from '@emotion/styled';

import { Theme } from 'verdaccio-ui/design-tokens/theme';

import ListItem from '../ListItem';
import ListItemText from '../ListItemText';
import Text from '../Text';

export const StyledText = styled(Text)<{ theme?: Theme }>(props => ({
  fontWeight: props.theme && props.theme.fontWeight.bold,
}));

export const AuthorListItem = styled(ListItem)({
  padding: 0,
  ':hover': {
    backgroundColor: 'transparent',
  },
});

export const AuthorListItemText = styled(ListItemText)({
  padding: '0 10px',
  margin: 0,
});
