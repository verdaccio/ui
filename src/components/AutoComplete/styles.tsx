import styled from '@emotion/styled';
import React from 'react';

import { Theme } from 'verdaccio-ui/design-tokens/theme';

import Paper from '../Paper';
import TextField from '../TextField';

export interface InputFieldProps {
  color: string;
}

export const Wrapper = styled('div')({
  width: '100%',
  height: '32px',
  position: 'relative',
  zIndex: 1,
});

export const StyledTextField = styled(TextField)<{ theme?: Theme }>(props => ({
  '& .MuiInputBase-root': {
    ':before': {
      content: "''",
      border: 'none',
    },
    ':after': {
      borderColor: props.theme && props.theme.palette.white,
    },
    ':hover:before': {
      content: 'none',
    },
  },
  '& .MuiInputBase-input': {
    color: props.theme && props.theme.palette.white,
  },
}));

/* eslint-disable verdaccio/jsx-spread */
// @ts-ignore types of color are incompatible
export const InputField: React.FC<InputFieldProps> = ({ ...others }) => <StyledTextField {...others} />;

export const SuggestionContainer = styled(Paper)({
  maxHeight: '500px',
  overflowY: 'auto',
});
