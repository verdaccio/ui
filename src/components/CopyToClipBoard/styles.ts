import styled from '@emotion/styled';

import IconButton from '../IconButton';

export const ClipBoardCopy = styled('div')({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'space-between',
});

export const ClipBoardCopyText = styled('span')({
  display: 'inline-block',
  textOverflow: 'ellipsis',
  overflow: 'hidden',
  whiteSpace: 'nowrap',
  height: '21px',
  fontSize: '1rem',
});

export const CopyIcon = styled(IconButton)({});
