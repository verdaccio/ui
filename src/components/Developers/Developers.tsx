import React, { useState, useCallback, useContext, useEffect, useMemo } from 'react';
import Add from '@material-ui/icons/Add';
import styled from '@emotion/styled';

import { DetailContext } from '../../pages/Version';
import Tooltip from '../../muiComponents/Tooltip';
import Avatar from '../../muiComponents/Avatar';
import Box from '../../muiComponents/Box';
import Text from '../../muiComponents/Text';
import FloatingActionButton from '../../muiComponents/FloatingActionButton';
import { Theme } from '../../design-tokens/theme';

import getUniqueDeveloperValues from './get-unique-developer-values';

export const Fab = styled(FloatingActionButton)<{ theme?: Theme }>(props => ({
  backgroundColor: props.theme && props.theme.palette.primary.main,
  color: props.theme && props.theme.palette.white,
}));

export enum DeveloperType {
  CONTRIBUTORS = 'contributors',
  MAINTAINERS = 'maintainers',
}

interface Props {
  type: DeveloperType;
  visibleMax?: number;
}

export const StyledText = styled(Text)<{ theme?: Theme }>(({ theme }) => ({
  fontWeight: theme && theme.fontWeight.bold,
  marginBottom: '10px',
  textTransform: 'capitalize',
}));

const StyledBox = styled(Box)({
  '> *': {
    margin: 5,
  },
});

export const VISIBLE_MAX = 2;

const Developers: React.FC<Props> = ({ type, visibleMax = VISIBLE_MAX }) => {
  const detailContext = useContext(DetailContext);

  if (!detailContext) {
    throw Error("The app's detail Context was not correct used");
  }

  const developers = useMemo(() => getUniqueDeveloperValues(detailContext.packageMeta?.latest[type]), [
    detailContext.packageMeta,
    type,
  ]);

  const [visibleDevelopersMax, setVisibleDevelopersMax] = useState(visibleMax);
  const [visibleDevelopers, setVisibleDevelopers] = useState(developers);

  useEffect(() => {
    if (!developers) return;
    setVisibleDevelopers(developers.slice(0, visibleDevelopersMax));
  }, [developers, visibleDevelopersMax]);

  const handleSetVisibleDevelopersMax = useCallback(() => {
    setVisibleDevelopersMax(visibleDevelopersMax + VISIBLE_MAX);
  }, [visibleDevelopersMax]);

  if (!visibleDevelopers || !developers) return null;

  return (
    <>
      <StyledText variant={'subtitle1'}>{type}</StyledText>
      <StyledBox display="flex" flexWrap="wrap" margin="10px 0 10px 0">
        {visibleDevelopers.map(visibleDeveloper => (
          <Tooltip key={visibleDeveloper.email} title={visibleDeveloper.name}>
            <Avatar alt={visibleDeveloper.name} src={visibleDeveloper.avatar} />
          </Tooltip>
        ))}
        {visibleDevelopersMax < developers.length && (
          <Fab onClick={handleSetVisibleDevelopersMax} size="small">
            <Add />
          </Fab>
        )}
      </StyledBox>
    </>
  );
};

export default Developers;
