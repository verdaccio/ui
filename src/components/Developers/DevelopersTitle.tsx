import React from 'react';
import styled from '@emotion/styled';
import { useTranslation } from 'react-i18next';

import Text from '../../muiComponents/Text';
import { Theme } from '../../design-tokens/theme';

import { DeveloperType } from './types';

interface Props {
  type: DeveloperType;
}

const DevelopersTitle: React.FC<Props> = ({ type }) => {
  const { t } = useTranslation();
  switch (type) {
    case DeveloperType.CONTRIBUTORS:
      return <StyledText variant={'subtitle1'}>{t('sidebar.contributors.title')}</StyledText>;
    case DeveloperType.MAINTAINERS:
      return <StyledText variant={'subtitle1'}>{t('sidebar.maintainers.title')}</StyledText>;
      return null;
  }
};

export default DevelopersTitle;

const StyledText = styled(Text)<{ theme?: Theme }>(({ theme }) => ({
  fontWeight: theme && theme.fontWeight.bold,
  marginBottom: '10px',
}));
