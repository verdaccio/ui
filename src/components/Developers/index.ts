export { default, DeveloperType } from './Developers';
