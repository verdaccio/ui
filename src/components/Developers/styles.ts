import styled from 'react-emotion';
import { default as MuiFab } from '@material-ui/core/Fab';

import colors from '../../utils/styles/colors';
import { fontWeight } from '../../utils/styles/sizes';
import Text from '../../muiComponents/Text';

export const Details = styled('span')({
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
});

export const Content = styled('div')({
  margin: '10px 0 10px 0',
  display: 'flex',
  flexWrap: 'wrap',
  '> *': {
    margin: '5px',
  },
});

export const StyledText = styled(Text)({
  fontWeight: fontWeight.bold,
  marginBottom: '10px',
  textTransform: 'capitalize',
});

export const Fab = styled(MuiFab)({
  '&&': {
    backgroundColor: colors.primary,
    color: colors.white,
  },
});
