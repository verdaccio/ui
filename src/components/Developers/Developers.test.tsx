import React from 'react';
import { mount } from 'enzyme';
import Developers, { DevelopersType } from './Developers';
import { Fab } from './styles';
import { DetailContextProvider } from '../../pages/Version';

describe('test Developers', () => {
  const packageMeta = {
    latest: {
      packageName: 'foo',
      version: '1.0.0',
      maintainers: [
        {
          name: 'dmethvin',
          email: 'dave.methvin@gmail.com',
        },
        {
          name: 'mgol',
          email: 'm.goleb@gmail.com',
        },
      ],
      contributors: [
        {
          name: 'dmethvin',
          email: 'dave.methvin@gmail.com',
        },
        {
          name: 'mgol',
          email: 'm.goleb@gmail.com',
        },
      ],
    },
  };

  test('should render the component with no items', () => {
    const type: DevelopersType = 'maintainers';
    const packageMeta = {
      latest: {},
    };
    const wrapper = mount(
      // @ts-ignore
      <DetailContextProvider value={{ packageMeta }}>
        <Developers type={type} />
      </DetailContextProvider>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('should render the component for maintainers with items', () => {
    const type: DevelopersType = 'maintainers';
    const wrapper = mount(
      // @ts-ignore
      <DetailContextProvider value={{ packageMeta }}>
        <Developers type={type} />
      </DetailContextProvider>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('should render the component for contributors with items', () => {
    const type: DevelopersType = 'contributors';
    const wrapper = mount(
      // @ts-ignore
      <DetailContextProvider value={{ packageMeta }}>
        <Developers type={type} />
      </DetailContextProvider>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('should test onClick the component avatar', () => {
    const type: DevelopersType = 'contributors';
    const packageMeta = {
      latest: {
        packageName: 'foo',
        version: '1.0.0',
        contributors: [
          {
            name: 'dmethvin',
            email: 'dave.methvin@gmail.com',
          },
          {
            name: 'dmethvin2',
            email: 'dave2.methvin@gmail.com',
          },
        ],
      },
    };

    const wrapper = mount(
      // @ts-ignore
      <DetailContextProvider value={{ packageMeta }}>
        <Developers type={type} visibleMax={1} />
      </DetailContextProvider>
    );

    const item2 = wrapper.find(Fab);
    // TODO: I am not sure here how to verify the method inside the component was called.
    item2.simulate('click');
  });
});
