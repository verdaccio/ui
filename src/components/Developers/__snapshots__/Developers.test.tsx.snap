// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Developers should render the component for contributors with items 1`] = `
<Developers
  type="contributors"
>
  <Styled(Component)
    variant="subtitle1"
  >
    <ForwardRef(Text)
      className="css-48zeoi emotion-0"
      variant="subtitle1"
    >
      <WithStyles(ForwardRef(Typography))
        className="css-48zeoi emotion-0"
        variant="subtitle1"
      >
        <ForwardRef(Typography)
          className="css-48zeoi emotion-0"
          classes={
            Object {
              "alignCenter": "MuiTypography-alignCenter",
              "alignJustify": "MuiTypography-alignJustify",
              "alignLeft": "MuiTypography-alignLeft",
              "alignRight": "MuiTypography-alignRight",
              "body1": "MuiTypography-body1",
              "body2": "MuiTypography-body2",
              "button": "MuiTypography-button",
              "caption": "MuiTypography-caption",
              "colorError": "MuiTypography-colorError",
              "colorInherit": "MuiTypography-colorInherit",
              "colorPrimary": "MuiTypography-colorPrimary",
              "colorSecondary": "MuiTypography-colorSecondary",
              "colorTextPrimary": "MuiTypography-colorTextPrimary",
              "colorTextSecondary": "MuiTypography-colorTextSecondary",
              "displayBlock": "MuiTypography-displayBlock",
              "displayInline": "MuiTypography-displayInline",
              "gutterBottom": "MuiTypography-gutterBottom",
              "h1": "MuiTypography-h1",
              "h2": "MuiTypography-h2",
              "h3": "MuiTypography-h3",
              "h4": "MuiTypography-h4",
              "h5": "MuiTypography-h5",
              "h6": "MuiTypography-h6",
              "noWrap": "MuiTypography-noWrap",
              "overline": "MuiTypography-overline",
              "paragraph": "MuiTypography-paragraph",
              "root": "MuiTypography-root",
              "srOnly": "MuiTypography-srOnly",
              "subtitle1": "MuiTypography-subtitle1",
              "subtitle2": "MuiTypography-subtitle2",
            }
          }
          variant="subtitle1"
        >
          <h6
            className="MuiTypography-root css-48zeoi emotion-0 MuiTypography-subtitle1"
          >
            contributors
          </h6>
        </ForwardRef(Typography)>
      </WithStyles(ForwardRef(Typography))>
    </ForwardRef(Text)>
  </Styled(Component)>
  <Styled(div)>
    <div
      className="css-mkcn9c emotion-6"
    >
      <Styled(span)
        key="dave.methvin@gmail.com"
      >
        <span
          className="css-dvxtzn emotion-4"
        >
          <AvatarTooltip
            email="dave.methvin@gmail.com"
            name="dmethvin"
            version="1.0.0"
          >
            <ForwardRef(ToolTip)
              title="dmethvin"
            >
              <WithStyles(ForwardRef(Tooltip))
                innerRef={null}
                title="dmethvin"
              >
                <ForwardRef(Tooltip)
                  classes={
                    Object {
                      "popper": "MuiTooltip-popper",
                      "popperInteractive": "MuiTooltip-popperInteractive",
                      "tooltip": "MuiTooltip-tooltip",
                      "tooltipPlacementBottom": "MuiTooltip-tooltipPlacementBottom",
                      "tooltipPlacementLeft": "MuiTooltip-tooltipPlacementLeft",
                      "tooltipPlacementRight": "MuiTooltip-tooltipPlacementRight",
                      "tooltipPlacementTop": "MuiTooltip-tooltipPlacementTop",
                      "touch": "MuiTooltip-touch",
                    }
                  }
                  title="dmethvin"
                >
                  <a
                    aria-describedby={null}
                    className=""
                    href="mailto:dave.methvin@gmail.com?subject=undefined@1.0.0"
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOver={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    target="_top"
                    title="dmethvin"
                  >
                    <ForwardRef(Avatar)
                      aria-label="dmethvin"
                    >
                      <WithStyles(ForwardRef(Avatar))
                        aria-label="dmethvin"
                      >
                        <ForwardRef(Avatar)
                          aria-label="dmethvin"
                          classes={
                            Object {
                              "colorDefault": "MuiAvatar-colorDefault",
                              "img": "MuiAvatar-img",
                              "root": "MuiAvatar-root",
                            }
                          }
                        >
                          <div
                            aria-label="dmethvin"
                            className="MuiAvatar-root MuiAvatar-colorDefault"
                          />
                        </ForwardRef(Avatar)>
                      </WithStyles(ForwardRef(Avatar))>
                    </ForwardRef(Avatar)>
                  </a>
                  <ForwardRef(Popper)
                    anchorEl={
                      <a
                        class=""
                        href="mailto:dave.methvin@gmail.com?subject=undefined@1.0.0"
                        target="_top"
                        title="dmethvin"
                      >
                        <div
                          aria-label="dmethvin"
                          class="MuiAvatar-root MuiAvatar-colorDefault"
                        />
                      </a>
                    }
                    className="MuiTooltip-popper"
                    id={null}
                    open={false}
                    placement="bottom"
                    transition={true}
                  />
                </ForwardRef(Tooltip)>
              </WithStyles(ForwardRef(Tooltip))>
            </ForwardRef(ToolTip)>
          </AvatarTooltip>
        </span>
      </Styled(span)>
      <Styled(span)
        key="m.goleb@gmail.com"
      >
        <span
          className="css-dvxtzn emotion-4"
        >
          <AvatarTooltip
            email="m.goleb@gmail.com"
            name="mgol"
            version="1.0.0"
          >
            <ForwardRef(ToolTip)
              title="mgol"
            >
              <WithStyles(ForwardRef(Tooltip))
                innerRef={null}
                title="mgol"
              >
                <ForwardRef(Tooltip)
                  classes={
                    Object {
                      "popper": "MuiTooltip-popper",
                      "popperInteractive": "MuiTooltip-popperInteractive",
                      "tooltip": "MuiTooltip-tooltip",
                      "tooltipPlacementBottom": "MuiTooltip-tooltipPlacementBottom",
                      "tooltipPlacementLeft": "MuiTooltip-tooltipPlacementLeft",
                      "tooltipPlacementRight": "MuiTooltip-tooltipPlacementRight",
                      "tooltipPlacementTop": "MuiTooltip-tooltipPlacementTop",
                      "touch": "MuiTooltip-touch",
                    }
                  }
                  title="mgol"
                >
                  <a
                    aria-describedby={null}
                    className=""
                    href="mailto:m.goleb@gmail.com?subject=undefined@1.0.0"
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOver={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    target="_top"
                    title="mgol"
                  >
                    <ForwardRef(Avatar)
                      aria-label="mgol"
                    >
                      <WithStyles(ForwardRef(Avatar))
                        aria-label="mgol"
                      >
                        <ForwardRef(Avatar)
                          aria-label="mgol"
                          classes={
                            Object {
                              "colorDefault": "MuiAvatar-colorDefault",
                              "img": "MuiAvatar-img",
                              "root": "MuiAvatar-root",
                            }
                          }
                        >
                          <div
                            aria-label="mgol"
                            className="MuiAvatar-root MuiAvatar-colorDefault"
                          />
                        </ForwardRef(Avatar)>
                      </WithStyles(ForwardRef(Avatar))>
                    </ForwardRef(Avatar)>
                  </a>
                  <ForwardRef(Popper)
                    anchorEl={
                      <a
                        class=""
                        href="mailto:m.goleb@gmail.com?subject=undefined@1.0.0"
                        target="_top"
                        title="mgol"
                      >
                        <div
                          aria-label="mgol"
                          class="MuiAvatar-root MuiAvatar-colorDefault"
                        />
                      </a>
                    }
                    className="MuiTooltip-popper"
                    id={null}
                    open={false}
                    placement="bottom"
                    transition={true}
                  />
                </ForwardRef(Tooltip)>
              </WithStyles(ForwardRef(Tooltip))>
            </ForwardRef(ToolTip)>
          </AvatarTooltip>
        </span>
      </Styled(span)>
    </div>
  </Styled(div)>
</Developers>
`;

exports[`test Developers should render the component for maintainers with items 1`] = `
<Developers
  type="maintainers"
>
  <Styled(Component)
    variant="subtitle1"
  >
    <ForwardRef(Text)
      className="css-48zeoi emotion-0"
      variant="subtitle1"
    >
      <WithStyles(ForwardRef(Typography))
        className="css-48zeoi emotion-0"
        variant="subtitle1"
      >
        <ForwardRef(Typography)
          className="css-48zeoi emotion-0"
          classes={
            Object {
              "alignCenter": "MuiTypography-alignCenter",
              "alignJustify": "MuiTypography-alignJustify",
              "alignLeft": "MuiTypography-alignLeft",
              "alignRight": "MuiTypography-alignRight",
              "body1": "MuiTypography-body1",
              "body2": "MuiTypography-body2",
              "button": "MuiTypography-button",
              "caption": "MuiTypography-caption",
              "colorError": "MuiTypography-colorError",
              "colorInherit": "MuiTypography-colorInherit",
              "colorPrimary": "MuiTypography-colorPrimary",
              "colorSecondary": "MuiTypography-colorSecondary",
              "colorTextPrimary": "MuiTypography-colorTextPrimary",
              "colorTextSecondary": "MuiTypography-colorTextSecondary",
              "displayBlock": "MuiTypography-displayBlock",
              "displayInline": "MuiTypography-displayInline",
              "gutterBottom": "MuiTypography-gutterBottom",
              "h1": "MuiTypography-h1",
              "h2": "MuiTypography-h2",
              "h3": "MuiTypography-h3",
              "h4": "MuiTypography-h4",
              "h5": "MuiTypography-h5",
              "h6": "MuiTypography-h6",
              "noWrap": "MuiTypography-noWrap",
              "overline": "MuiTypography-overline",
              "paragraph": "MuiTypography-paragraph",
              "root": "MuiTypography-root",
              "srOnly": "MuiTypography-srOnly",
              "subtitle1": "MuiTypography-subtitle1",
              "subtitle2": "MuiTypography-subtitle2",
            }
          }
          variant="subtitle1"
        >
          <h6
            className="MuiTypography-root css-48zeoi emotion-0 MuiTypography-subtitle1"
          >
            maintainers
          </h6>
        </ForwardRef(Typography)>
      </WithStyles(ForwardRef(Typography))>
    </ForwardRef(Text)>
  </Styled(Component)>
  <Styled(div)>
    <div
      className="css-mkcn9c emotion-6"
    >
      <Styled(span)
        key="dave.methvin@gmail.com"
      >
        <span
          className="css-dvxtzn emotion-4"
        >
          <AvatarTooltip
            email="dave.methvin@gmail.com"
            name="dmethvin"
            version="1.0.0"
          >
            <ForwardRef(ToolTip)
              title="dmethvin"
            >
              <WithStyles(ForwardRef(Tooltip))
                innerRef={null}
                title="dmethvin"
              >
                <ForwardRef(Tooltip)
                  classes={
                    Object {
                      "popper": "MuiTooltip-popper",
                      "popperInteractive": "MuiTooltip-popperInteractive",
                      "tooltip": "MuiTooltip-tooltip",
                      "tooltipPlacementBottom": "MuiTooltip-tooltipPlacementBottom",
                      "tooltipPlacementLeft": "MuiTooltip-tooltipPlacementLeft",
                      "tooltipPlacementRight": "MuiTooltip-tooltipPlacementRight",
                      "tooltipPlacementTop": "MuiTooltip-tooltipPlacementTop",
                      "touch": "MuiTooltip-touch",
                    }
                  }
                  title="dmethvin"
                >
                  <a
                    aria-describedby={null}
                    className=""
                    href="mailto:dave.methvin@gmail.com?subject=undefined@1.0.0"
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOver={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    target="_top"
                    title="dmethvin"
                  >
                    <ForwardRef(Avatar)
                      aria-label="dmethvin"
                    >
                      <WithStyles(ForwardRef(Avatar))
                        aria-label="dmethvin"
                      >
                        <ForwardRef(Avatar)
                          aria-label="dmethvin"
                          classes={
                            Object {
                              "colorDefault": "MuiAvatar-colorDefault",
                              "img": "MuiAvatar-img",
                              "root": "MuiAvatar-root",
                            }
                          }
                        >
                          <div
                            aria-label="dmethvin"
                            className="MuiAvatar-root MuiAvatar-colorDefault"
                          />
                        </ForwardRef(Avatar)>
                      </WithStyles(ForwardRef(Avatar))>
                    </ForwardRef(Avatar)>
                  </a>
                  <ForwardRef(Popper)
                    anchorEl={
                      <a
                        class=""
                        href="mailto:dave.methvin@gmail.com?subject=undefined@1.0.0"
                        target="_top"
                        title="dmethvin"
                      >
                        <div
                          aria-label="dmethvin"
                          class="MuiAvatar-root MuiAvatar-colorDefault"
                        />
                      </a>
                    }
                    className="MuiTooltip-popper"
                    id={null}
                    open={false}
                    placement="bottom"
                    transition={true}
                  />
                </ForwardRef(Tooltip)>
              </WithStyles(ForwardRef(Tooltip))>
            </ForwardRef(ToolTip)>
          </AvatarTooltip>
        </span>
      </Styled(span)>
      <Styled(span)
        key="m.goleb@gmail.com"
      >
        <span
          className="css-dvxtzn emotion-4"
        >
          <AvatarTooltip
            email="m.goleb@gmail.com"
            name="mgol"
            version="1.0.0"
          >
            <ForwardRef(ToolTip)
              title="mgol"
            >
              <WithStyles(ForwardRef(Tooltip))
                innerRef={null}
                title="mgol"
              >
                <ForwardRef(Tooltip)
                  classes={
                    Object {
                      "popper": "MuiTooltip-popper",
                      "popperInteractive": "MuiTooltip-popperInteractive",
                      "tooltip": "MuiTooltip-tooltip",
                      "tooltipPlacementBottom": "MuiTooltip-tooltipPlacementBottom",
                      "tooltipPlacementLeft": "MuiTooltip-tooltipPlacementLeft",
                      "tooltipPlacementRight": "MuiTooltip-tooltipPlacementRight",
                      "tooltipPlacementTop": "MuiTooltip-tooltipPlacementTop",
                      "touch": "MuiTooltip-touch",
                    }
                  }
                  title="mgol"
                >
                  <a
                    aria-describedby={null}
                    className=""
                    href="mailto:m.goleb@gmail.com?subject=undefined@1.0.0"
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOver={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    target="_top"
                    title="mgol"
                  >
                    <ForwardRef(Avatar)
                      aria-label="mgol"
                    >
                      <WithStyles(ForwardRef(Avatar))
                        aria-label="mgol"
                      >
                        <ForwardRef(Avatar)
                          aria-label="mgol"
                          classes={
                            Object {
                              "colorDefault": "MuiAvatar-colorDefault",
                              "img": "MuiAvatar-img",
                              "root": "MuiAvatar-root",
                            }
                          }
                        >
                          <div
                            aria-label="mgol"
                            className="MuiAvatar-root MuiAvatar-colorDefault"
                          />
                        </ForwardRef(Avatar)>
                      </WithStyles(ForwardRef(Avatar))>
                    </ForwardRef(Avatar)>
                  </a>
                  <ForwardRef(Popper)
                    anchorEl={
                      <a
                        class=""
                        href="mailto:m.goleb@gmail.com?subject=undefined@1.0.0"
                        target="_top"
                        title="mgol"
                      >
                        <div
                          aria-label="mgol"
                          class="MuiAvatar-root MuiAvatar-colorDefault"
                        />
                      </a>
                    }
                    className="MuiTooltip-popper"
                    id={null}
                    open={false}
                    placement="bottom"
                    transition={true}
                  />
                </ForwardRef(Tooltip)>
              </WithStyles(ForwardRef(Tooltip))>
            </ForwardRef(ToolTip)>
          </AvatarTooltip>
        </span>
      </Styled(span)>
    </div>
  </Styled(div)>
</Developers>
`;

exports[`test Developers should render the component with no items 1`] = `
<Developers
  type="maintainers"
/>
`;
